common/spec/dependabot/file_fetchers/base_spec.rb

```diff
@@ -1313,4 +1319,82 @@ def fetch_files
       end
     end
   end
+
+  context "with repo_contents_path" do
+    let(:repo_contents_path) { Dir.mktmpdir }
+    # prepare a checkout to spoof `git clone`
+    let(:contents) { fixture("ruby", "gemfiles", "Gemfile") }
+    let(:mock_clone) do
+      `git init #{repo_contents_path}`
+      path = File.join(repo_contents_path, "requirements.txt")
+      File.write(path, contents)
+      `git -C #{repo_contents_path} add .`
+      `git -C #{repo_contents_path} commit -m'fake clone'`
+      repo_contents_path
+    end
```

Comment:
From what I can tell, it seems like we're mostly using this let block for its side effects, as opposed to using it to set a variable. With that in mind, I recommend using a [`before(:example)`](https://rspec.info/documentation/3.9/rspec-core/RSpec/Core/Hooks.html#before-instance_method) instead:

```suggestion
    before(:example) do
      `git init #{repo_contents_path}`
      path = File.join(repo_contents_path, "requirements.txt")
      File.write(path, contents)
      `git -C #{repo_contents_path} add .`
      `git -C #{repo_contents_path} commit -m'fake clone'`
    end
```

That would also allow us to remove the `repo_contents_path` return value. What do you think?

Reply:
I'd assumed `mock_clone` would be lazily evaluated, so `and_return(mock_clone)` below would mean this checkout doesn't exist until `FileFetcher#_clone_repo_contents` is called.

Unconfirmed assumption, but it felt pretty slick.

Reply:
I think this concern (relying on the side-effect) is 💯 in the latest rev.